src/components/signup/SignupForm/index.js

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Redirect, useHistory } from 'react-router-dom';
+import { z } from 'zod';
+
+import Input from 'components/form/Input';
+import Button from 'components/common/Button';
+import routesPaths from 'routes/routesPaths';
+import useTranslation from 'hooks/useTranslation';
+import useAuth from 'hooks/useAuth';
+import { api } from 'services/api';
+import { useSignupMutation } from 'services/auth/auth';
+import { setLoggedInUser } from 'utils/auth';
+import { PASSWORD_REGEX } from 'constants/constants';
+import './styles.scss';
+import Select from 'components/form/Select';
+
+const SignupForm = () => {
+  const t = useTranslation();
+  const dispatch = useDispatch();
+  const { push } = useHistory();
+  const { user, authenticated } = useAuth();
+  const [signup, { isLoading, isSuccess, error }] = useSignupMutation();
+
+  const schema = z
+    .object({
+      name: z.string(),
+      gender: z.string(),
+      email: z.string().email({ message: t('signup.errors.emailMsg') }),
+      password: z.string().regex(PASSWORD_REGEX, { message: t('signup.errors.passwordMsg') }),
+      passwordConfirmation: z
+        .string()
+        .regex(PASSWORD_REGEX, { message: t('signup.errors.passwordMsg') }),
+    })
+    .refine(data => data.password === data.passwordConfirmation, {
+      message: t('signup.errors.passwordConfirmationMsg'),
+      path: ['passwordConfirmation'],
+    });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: zodResolver(schema) });
+
+  const onSubmit = data => signup(data);
+
+  const resetErrors = useCallback(() => dispatch(api.util.resetApiState()), [dispatch]);
+
+  const handleFocus = () => error && resetErrors();
+
+  useEffect(() => {
+    if (isSuccess) {
+      setLoggedInUser(user);
+      push(routesPaths.index);
+    }
+  }, [isSuccess, user, push]);
+
+  useEffect(() => resetErrors, [resetErrors]);
+
+  if (authenticated) {
+    return <Redirect to={routesPaths.index} />;
+  }
+
+  return (
```

Comment:
you can create a custom hook to remove logic from this component as it is getting large